File: output/ChatGPT/avatar/Python/C++/codeforces_518_B.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main() {
    string s, t;
    cin >> s >> t;
    
    unordered_map<char, int> freqs;
    unordered_map<char, int> freqt;
    int yay = 0, whoops = 0;
    
    for (char letra : s) {
        freqs[letra]++;
    }
    
    for (char letra : t) {
        freqt[letra]++;
    }
    
    for (char letra : s) {
        if (freqs[letra] > 0 && freqt[letra] > 0) {
            yay++;
            freqs[letra]--;
            freqt[letra]--;
        }
    }
    
    for (char letra : s) {
        if (freqs[letra] > 0) {
            if (islower(letra) && freqt[toupper(letra)] > 0) {
                whoops++;
                freqs[letra]--;
                freqt[toupper(letra)]--;
            }
            else if (isupper(letra) && freqt[tolower(letra)] > 0) {
                whoops++;
                freqs[letra]--;
                freqt[tolower(letra)]--;
            }
        }
    }
    
    cout << yay << " " << whoops << endl;
    
    return 0;
}

File: output/CodeGen/avatar/Python/Java_handraft/codeforces_518_B.java
public class codeforces_518_B {
    public int numSimilarGroups(String[] A, String[] B) {
        int yay = 0;
        int whoops = 0;
        HashMap<Character, Integer> freqs = new HashMap<Character, Integer>();
        HashMap<Character, Integer> freqt = new HashMap<Character, Integer>();
        for (String s : A) {
            for (char letra : s.toCharArray()) {
                if (freqs.containsKey(letra)) {
                    freqs.put(letra, freqs.get(letra) + 1);
                } else {
                    freqs.put(letra, 1);
                }
            }
        }
        for (String t : B) {
            for (char letra : t.toCharArray()) {
                if (freqt.containsKey(letra)) {
                    freqt.put(letra, freqt.get(letra) + 1);
                } else {
                    freqt.put(letra, 1);
                }
            }
        }
        for (char letra : freqs.keySet()) {
            while (freqs.get(letra) > 0 && letra in freqt && freqt.get(letra) > 0) {
                yay++;
                freqs.put(letra, freqs.get(letra) - 1);
                freqt.put(letra, freqt.get(letra) - 1);
            }
        }
        for (char letra : freqs.keySet()) {
            while (freqs.get(letra) > 0) {
                if (letra.isLowerCase() && letra.toUpperCase() in freqt.keySet() && freqt.get(letra.toUpperCase()) > 0) {
                    whoops++;
                    freqs.put(letra, freqs.get(letra) - 1);
                    freqt.put(letra.toUpperCase(), freqt.get(letra.toUpperCase()) - 1);
                } else if (letra.isUpperCase() && letra.toLowerCase() in freqt.keySet() && freqt.get(letra.toLowerCase()) > 0) {
                    whoops++;
                    freqs.put(letra, freqs.get(letra) - 1);
                    freqt.put(letra.toLowerCase(), freqt.get(letra.toLowerCase()) - 1);
                } else {
                    break;
                }
            }
        }
        return yay + whoops;
    }
}